src/Lecture2.hs

```diff
@@ -97,7 +110,11 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+dropSpaces :: [Char] -> [Char]
+dropSpaces = head . words
+-- FIRST ATTEMPT
+-- choked on word with infinite amount of spaces on the right hand side
+-- dropSpaces = filter (not . isSpace)
```

Comment:
That's a clever solution that relies on Haskell laziness 😅 

Alternatively, you can implement it like this:

```haskell
dropSpaces = takeWhile (not . isSpace) . dropWhile isSpace
```